components/YarnStockForm/index.js

```diff
@@ -0,0 +1,115 @@
+import styled from "styled-components";
+import { ButtonContainer } from "../../styles";
+import Heading from "../Heading";
+import StyledInput from "../StyledInput";
+import StyledButton from "../StyledButton";
+import BackIcon from "../Icon/BackIcon";
+import useSWR from "swr";
+import { useRouter } from "next/router";
+
+export default function YarnStockForm({ setAddYarnStockStatus }) {
+  const { mutate } = useSWR("/api/project");
+  const router = useRouter();
+
+  async function handleAddYarnStockSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    const response = await fetch("/api/yarn", {
+      method: "POST",
+
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ data }),
+    });
+
+    router.push("/");
+    mutate();
+  }
```

Comment:
not sure if it is sth to mention, but maybe switch `router.push("/")` and `mutate()` to have the reRender happen before the user gets back to `"/"`.
But possible, that my thinking is wrong

Reply:
Hi i am also not sure about that, but it makes sense that first mutate then push to other page. i will change it by next PR thanks!